validate/validate.go

```diff
@@ -5,12 +5,103 @@ import (
 	"fmt"
 	"github.com/hashicorp/hcl/v2"
 	"io/ioutil"
+	"regexp"
 )
 
-// WordWithDotHyphenUnder80Pattern is a regexp pattern that matches string that contain alphanumerics, underscores, periods,
+type RegexpValidator struct {
+	pattern       string
+	errorTemplate string
+	regex         *regexp.Regexp
+}
+
+// Check validates provided string with a regexp based on the pattern and returns optional error.
+func (r *RegexpValidator) Check(value string, valueType interface{}) error {
```

Comment:
this should always be a string, no?

Reply:
Yeah, but I only put it in the formated string in the %v part, so AFAIK the type doesn't matter here.